src/main/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -693,6 +762,40 @@ private void addIndexAsDuplicate(final long bamIndex) {
         ++this.numDuplicateIndices;
     }
 
+    private void addRepresentativeReadOfDuplicateSet(final long representativeReadIndexInFile, final int setSize, final long read1IndexInFile) {
+        final RepresentativeReadIndexer rni = new RepresentativeReadIndexer();
+        rni.representativeReadIndexInFile = (int) representativeReadIndexInFile;
+        rni.setSize = setSize;
+        rni.readIndexInFile = (int) read1IndexInFile;
+        this.representativeReadIndicesForDuplicates.add(rni);
+    }
+
+    /**
+     * Takes a list of ReadEndsForMarkDuplicates objects and adds the representative read index
+     * for the first and second in a pair
+     *
+     * @param list
+     */
+    private void addRepresentativeReadIndex(final List<ReadEndsForMarkDuplicates> list) {
+        short maxScore = 0;
+        ReadEndsForMarkDuplicates best = null;
+
+        /** All read ends should have orientation FF, FR, RF, or RR **/
+        for (final ReadEndsForMarkDuplicates end : list) {
+            if (end.score > maxScore || best == null) {
+                maxScore = end.score;
+                best = end;
+            }
+        }
+
+        // for read name (for representative read name), add the last of the pair that was examined
+        for (final ReadEndsForMarkDuplicates end : list) {
+            addRepresentativeReadOfDuplicateSet(best.read1IndexInFile, list.size(), end.read1IndexInFile);
+            addRepresentativeReadOfDuplicateSet(best.read1IndexInFile, list.size(), end.read2IndexInFile);
```

Comment:
No, I think we should have only one value that marks all members of the duplicate set. I think the read1 index-in-file of the representative read is a reasonable choice. I think this is better than having two separate indexes for the first and second pair. I’ll be more explicit about this in the description of this function.